### DISCUSSION TITLE
Java implementation of OSQP

### DISCUSSION THREAD

**Discussion by Laksh-kit** [osqp/osqp#686] [Q&A] [answered]
Dear Community, 
I am reasearcher from Germany in the field of energy management and operation optimization. I have tried our the OSQP solvers in Matlab and it works great for my simulations. But my end goal is to implement in my Energy Management System which is running on JAVA. Do you know if a Java version of the solvers are already available, if not are there any ways of making the solver compatible for JAVA in the form of a package? 

Regards
Laksh 

Update 1: I found OSQP in [Maven repository](https://mvnrepository.com/artifact/org.renjin.cran/osqp), but the last update is in 2022. Not sure if it is still the updated version. Could anyone guide me here?

**Comment by imciner2** [comment, marked as answer]
Hi @Laksh-kit,

We don't currently have a maintained Java wrapper for the OSQP library, since no one on the development team uses it or understands how to wrap it. However, there was recently a group that actually rewrote the OSQP algorithm into a pure-Java implementation. They made it open source here: https://github.com/quantego/josqp.

**Reply by Laksh-kit** [reply]
That is already great. Thanks for the tip.
